web-client/src/presenter/actions/setDawsonHasUpdatedAction.ts

```diff
@@ -0,0 +1,24 @@
+import { BROADCAST_MESSAGES } from '@shared/business/entities/EntityConstants';
+import { state } from '@web-client/presenter/app.cerebral';
+
+export const setDawsonHasUpdatedAction = async ({
+  applicationContext,
+  get,
+  props,
+  store,
+}: ActionProps) => {
+  store.set(state.dawsonHasUpdated, true);
+
+  // Clear the refresh token interval since all subsequent requests until refresh will fail
+  const oldInterval = get(state.refreshTokenInterval);
+  clearInterval(oldInterval);
+  store.unset(state.refreshTokenInterval);
```

Comment:
I'm debating on whether to pull this block out as a separate action. I like that it is closely coupled to this action. I don't like that it is essentially a subaction.